### PR TITLE
Consistently use URLs instead of str/URIs

### LINKED ISSUE #6: url-dir can return strings, despite its docstring
The docstring for `url-dir` says:
> Return a list of URLs contained by this URL, if the protocol supports it.

But `url-dir` method for `"jar"` uses `build-url` which returns strings.

### PR DESCRIPTION
**Pull request by p-himik**
Fixes #6

### CODE REVIEW

File: src/resauce/core.clj [hunk 1]
```diff
@@ -12,15 +12,20 @@
...
 
 (defn- build-url [base-url dir path]
-  (if (.startsWith ^String path dir)
-    (str (add-ending-slash (str base-url))
-         (subs path (count (add-ending-slash dir))))))
+  (assert (.startsWith ^String path dir))
+  (URL. (str (add-ending-slash (str base-url))
+             (subs path (count (add-ending-slash dir))))))
+
+(defn- ^URL ensure-url [url]
```
Comment (weavejester) [inline src/resauce/core.clj]: Why not use `clojure.java.io/as-url` instead?

File: src/resauce/core.clj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (ns resauce.core
   (:require [clojure.java.io :as io])
   (:import [java.io File]
-           [java.net JarURLConnection URI URL]
+           [java.net JarURLConnection URL URI]
```
Comment (weavejester) [inline src/resauce/core.clj]: This change looks like it's superfluous now.

File: src/resauce/core.clj [hunk 3]
```diff
@@ -12,15 +12,16 @@
     (filter (partial re-matches re) paths)))
 
 (defn- build-url [base-url dir path]
-  (if (.startsWith ^String path dir)
-    (str (add-ending-slash (str base-url))
-         (subs path (count (add-ending-slash dir))))))
+  (assert (.startsWith ^String path dir))
```
Comment (weavejester) [inline src/resauce/core.clj]: Why was this changed to an `assert`?

File: src/resauce/core.clj (hunk 2 above)
Comment (p-himik) [reply src/resauce/core.clj]: Right, missed that.

File: src/resauce/core.clj (hunk 3 above)
Comment (p-himik) [reply src/resauce/core.clj]: Because `build-url` is a private function that's only called in a context where `path` _must_ start with `dir`. Having a condition in there makes it seem like `url-dir` could return a list of `nil`s.

Comment (weavejester) [reply src/resauce/core.clj]: Okay. Let's make it a precondition then.
